flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/mailbox/execution/MailboxExecutor.java

```diff
@@ -91,4 +122,30 @@
 	 * @return only true if called from the mailbox thread.
 	 */
 	boolean isMailboxThread();
+
+	/**
+	 * Creates a view on the executor that adds a constant {@code toString()} to submitted executables.
+	 *
+	 * <p>This method should be used to provide a debuggable name when requiring an {@link Executor} (for example,
+	 * for {@link java.util.concurrent.CompletableFuture}).</p>
+	 *
+	 * @param name the bound name.
+	 * @return the executor view.
+	 */
+	default Executor asNamedExecutor(String name) {
+		return command -> MailboxExecutor.this.execute(namedRunnable(command, name));
+	}
+
+	/**
+	 * Creates a view on the executor that adds a constant {@code toString()} to submitted executables.
+	 *
+	 * <p>This method should be used to provide a debuggable name when requiring an {@link Executor} (for example,
+	 * for {@link java.util.concurrent.CompletableFuture}).</p>
+	 *
+	 * @param nameSupplier the lazily evaluated, bound name.
+	 * @return the executor view.
+	 */
+	default Executor asNamedExecutor(Supplier<String> nameSupplier) {
+		return command -> MailboxExecutor.this.execute(namedRunnable(command, nameSupplier));
```

Comment:
This and above method (`asNamedExecutor(String name)`) are not used?

Reply:
I added a test case now.